src/components/inputs/localized-rich-text-input/editor.js

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import { css } from '@emotion/core';
+import styled from '@emotion/styled';
+import { useIntl } from 'react-intl';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import usePrevious from '../../../hooks/use-previous';
+import CollapsibleMotion from '../../collapsible-motion';
+import Spacings from '../../spacings';
+import { AngleUpIcon } from '../../icons';
+import Text from '../../typography/text';
+import FlatButton from '../../buttons/flat-button';
+import RichTextBody from '../../internals/rich-text-body';
+import { getLanguageLabelStyles } from './editor.styles';
+import messages from './messages';
+
+const COLLAPSED_HEIGHT = 32;
+
+const LeftColumn = styled.div`
+  flex: 1;
+`;
+
+const RightColumn = styled.div`
+  flex: 0;
+`;
+
+const Row = styled.div`
+  display: flex;
+  &:empty {
+    margin: 0 !important;
+  }
+`;
+
+const Editor = props => {
+  const intl = useIntl();
+  const ref = React.useRef();
+  const prevIsFocused = usePrevious(props.editor.value.selection.isFocused);
+
+  const [renderToggleButton, setRenderToggleButton] = React.useState(false);
+
+  const { toggleLanguage } = props;
+  const onToggle = React.useCallback(() => {
+    toggleLanguage(props.language);
+  }, [toggleLanguage, props.language]);
+
+  const updateRenderToggleButton = React.useCallback(() => {
+    const doesExceedCollapsedHeightLimit =
+      ref.current.clientHeight > COLLAPSED_HEIGHT;
+
+    if (doesExceedCollapsedHeightLimit && !renderToggleButton) {
+      setRenderToggleButton(true);
+    }
+    if (!doesExceedCollapsedHeightLimit && renderToggleButton) {
+      setRenderToggleButton(false);
+    }
+  }, [setRenderToggleButton, renderToggleButton]);
+
+  React.useEffect(() => {
+    updateRenderToggleButton();
+  }, [props.editor.value.document, updateRenderToggleButton]);
```

Comment:
editor.value.document basically contains the entire value of the text editor, but not changes like focus or what no. So this useEffect runs pretty much whenever a user types / changes the contents of the editor.